lib/types.js

```diff
@@ -111,6 +111,7 @@ function Type(schema, opts) {
 Type.forSchema = function (schema, opts) {
   opts = opts || {};
   opts.registry = opts.registry || {};
+  opts.noStackTraces = opts.noStackTraces !== undefined ? opts.noStackTraces : false;
```

Comment:
Wasn't too sure where exactly to place this code, should the defaulting happen here on closer to where it used? I couldn't infer your style from the code.